lib-src/rpc2/include/rpc2/pack_helper.h

```diff
@@ -0,0 +1,88 @@
+/* BLURB lgpl
+
+                           Coda File System
+                              Release 5
+
+          Copyright (c) 1987-2016 Carnegie Mellon University
+                  Additional copyrights listed below
+
+This  code  is  distributed "AS IS" without warranty of any kind under
+the  terms of the  GNU  Library General Public Licence  Version 2,  as
+shown in the file LICENSE. The technical and financial contributors to
+Coda are listed in the file CREDITS.
+
+
+#*/
+
+
+#ifndef _PACK_HELPER_
+#define _PACK_HELPER_
+#include <rpc2/rpc2.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define STUBCLIENT 0
+#define STUBSERVER 1
+#define STUBIN 1
+#define STUBOUT 2
+#define _PAD(n)((((n)-1) | 3) + 1)
+typedef struct {
+    char* buffer;
+    char* eob;
+} BUFFER;
+
+
+int unpack_int(BUFFER *buf, RPC2_Integer *ptr);
+
+int unpack_unsigned(BUFFER *buf, RPC2_Unsigned *ptr);
+
+int unpack_double(BUFFER *buf, RPC2_Double *ptr);
+
+int unpack_bound_bytes(BUFFER *buf, unsigned char *ptr, RPC2_Unsigned len);
+
+int unpack_unbound_bytes(BUFFER *buf, unsigned char *ptr);
+
+int unpack_string(BUFFER *buf, unsigned char **ptr, int who);
```

Comment:
I think who, doesn't actually change and probably could be part of the BUFFER* structure. Not sure of mode at this point, I think that might change too often for it to be associated with the pack/unpack buffer.